pyomo/core/plugins/transform/constraint_tightener.py

```diff
@@ -0,0 +1,68 @@
+from pyomo.core.base.constraint import Constraint
+from pyomo.core.kernel.numvalue import value
+from pyomo.core.plugins.transform.hierarchy import IsomorphicTransformation
+from pyomo.repn.canonical_repn import generate_canonical_repn
+from pyomo.util.plugin import alias
+import logging
+import textwrap
+logger=logging.getLogger('pyomo.core')
+class TightenContraintFromVars(IsomorphicTransformation):
+    """Tightens bound on linear constraints, by iterating through each variable and
+    tightening the constraint bounds accordingly
+    """
+
+
+    alias('core.tighten_constraint_from_vars',doc=textwrap.fill(textwrap.dedent(__doc__.strip())))
+    def __init__(self):
+        super( TightenContraintFromVars, self).__init__()
+
+    def _apply_to(self,instance):
+
+        for constr in instance.component_data_objects(ctype = Constraint,
+                                                        active = True,
+                                                        descend_into=True):
+            if not constr.body.polynomial_degree() == 1:
+                continue
+
+            #tighten the constraint bound as much as possible
+            repn = generate_canonical_repn(constr.body)
+            UB = 0
+            if repn.constant:
+                UB = -1 * repn.constant
+            LB = UB
+            #loop through each coefficent and variable pair
+            for i, coef in enumerate(repn.linear):
+                #TODO: ROunding issues
+                #Calculate bounds using interval arithmetic
+                if coef >= 0:
+                    if repn.variables[i].ub:
+                        UB = UB + coef * value(repn.variables[i].ub)
+                    else:
+                        UB = float('Inf')
+                    if repn.variables[i].lb:
+                        LB = LB + coef * value(repn.variables[i].lb)
+                    else:
+                        LB = -float('Inf')
+                else:
+                    if repn.variables[i].ub:
```

Comment:
Should this be "if repn.variables[i].lb:"?

Reply:
No (and applies to comments later on), because coef is negative in these cases.